src/main/java/app/linguistai/bmvp/service/gamification/FriendshipService.java

```diff
@@ -0,0 +1,177 @@
+package app.linguistai.bmvp.service.gamification;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.UUID;
+
+import app.linguistai.bmvp.model.Friendship;
+import app.linguistai.bmvp.repository.gamification.IFriendshipRepository;
+
+import org.springframework.stereotype.Service;
+
+import app.linguistai.bmvp.model.User;
+import app.linguistai.bmvp.model.enums.FriendshipStatus;
+import app.linguistai.bmvp.repository.IAccountRepository;
+import jakarta.transaction.Transactional;
```

Comment:
Just fyi, using `org.springframework.transaction.annotation.Transactional` rather than `jakarta.transaction.Transactional` can be better for performance and other optimizations. So you can consider changing it but I think it is also okay if it stays like this as stated explained in the accepted answer [here (StackOverflow)](https://stackoverflow.com/questions/26387399/javax-transaction-transactional-vs-org-springframework-transaction-annotation-tr)